.github/workflows/build_and_test.yml

```diff
@@ -712,25 +712,15 @@ jobs:
     - name: Python code generation check
       if: inputs.branch != 'branch-3.3' && inputs.branch != 'branch-3.4'
       run: if test -f ./dev/connect-check-protos.py; then PATH=$PATH:$HOME/buf/bin PYTHON_EXECUTABLE=python3.9 ./dev/connect-check-protos.py; fi
-    - name: Install JavaScript linter dependencies
```

Comment:
After this pr merged, the daily tests of branch-3.x started to lack dependencies ...should similar optimizations all seem to be backported?

-branch-3.3 https://github.com/apache/spark/actions/runs/6918980555/job/18821819222
-branch-3.4 https://github.com/apache/spark/actions/runs/6919512513/job/18822994877
-branch-3.5 https://github.com/apache/spark/actions/runs/6920069263/job/18824167689

<img width="646" alt="image" src="https://github.com/apache/spark/assets/1475305/47cf950c-6c11-4bb0-9c1f-fa946faafd4d">

or should disable lint check for branch-3.x daily tests?

Reply:
Oh, too bad. I don't prefer backporing the infra PR, but I'll leave this to @zhengruifeng .

Reply:
oops, let me take a look

Reply:
it is hard to backport, I am going to restore installations in https://github.com/apache/spark/pull/43904